extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/metrics/MongoMetricsConnectionPoolListener.java

```diff
@@ -0,0 +1,95 @@
+import com.mongodb.event.ConnectionPoolOpenedEvent;
+import com.mongodb.event.ConnectionPoolWaitQueueEnteredEvent;
+import com.mongodb.event.ConnectionPoolWaitQueueExitedEvent;
+import com.mongodb.event.ConnectionRemovedEvent;
+
+import io.smallrye.metrics.MetricRegistries;
+
+public class MongoMetricsConnectionPoolListener implements ConnectionPoolListener {
+    private final static String sizeName = "mongodb.connection-pool.size";
+    private final static String checkedOutCountName = "mongodb.connection-pool.checked-out-count";
+
+    @Override
+    public void connectionPoolOpened(ConnectionPoolOpenedEvent event) {
+        Tag[] tags = createTags(event.getServerId());
+
+        Metadata sizeMetadata = Metadata.builder().withName(sizeName).withType(MetricType.COUNTER)
+                .withDescription("the current size of the pool, including idle and and in-use members").build();
+        Metadata checkedOutCountMetadata = Metadata.builder().withName(checkedOutCountName).withType(MetricType.COUNTER)
+                .withDescription(" the current count of connections that are currently in use").build();
+
+        getMetricRegistry().register(sizeMetadata, new ConnectionPoolCounter(), tags);
+        getMetricRegistry().register(checkedOutCountMetadata, new ConnectionPoolCounter(), tags);
+    }
+
+    @Override
+    public void connectionPoolClosed(ConnectionPoolClosedEvent event) {
+    }
+
+    @Override
+    public void connectionCheckedOut(ConnectionCheckedOutEvent event) {
+        Tag[] tags = createTags(event.getConnectionId().getServerId());
+
+        Counter checkedOutCount = getMetricRegistry().counter(checkedOutCountName, tags);
+        checkedOutCount.inc();
+    }
+
+    @Override
+    public void connectionCheckedIn(ConnectionCheckedInEvent event) {
+        Tag[] tags = createTags(event.getConnectionId().getServerId());
+
+        Counter checkedOutCount = getMetricRegistry().counter(checkedOutCountName, tags);
+        checkedOutCount.inc(-1);
+    }
+
+    @Override
+    public void waitQueueEntered(ConnectionPoolWaitQueueEnteredEvent connectionPoolWaitQueueEnteredEvent) {
+        // Not supported, since the event is deprecated and will be removed anyway
+    }
+
+    @Override
+    public void waitQueueExited(ConnectionPoolWaitQueueExitedEvent connectionPoolWaitQueueExitedEvent) {
+        // Not supported, since the event is deprecated and will be removed anyway
+    }
+
+    @Override
+    public void connectionAdded(ConnectionAddedEvent event) {
+        Tag[] tags = createTags(event.getConnectionId().getServerId());
+
+        Counter sizeCount = getMetricRegistry().counter(sizeName, tags);
+        sizeCount.inc();
+    }
+
+    @Override
+    public void connectionRemoved(ConnectionRemovedEvent event) {
+        Tag[] tags = createTags(event.getConnectionId().getServerId());
+
+        Counter sizeCount = getMetricRegistry().counter(sizeName, tags);
+        sizeCount.inc(-1);
```

Comment:
Counters are only supposed to go up. If you need a metric that goes down as well, it should be a `Gauge`. This would confuse Prometheus, which only expects counters to go up.